framework/decode/dx12_detection_consumer.h

```diff
@@ -50,11 +50,10 @@ class Dx12DetectionConsumer : public Dx12Consumer
 
     virtual bool IsComplete(uint64_t block_index) override
     {
-        return (block_index > kMaxDX12BlockLimit) || WasD3D12APIDetected();
+        return WasD3D12APIDetected();
```

Comment:
Instead of removing it, you could make `kMaxDX12BlockLimit` remain as a default, but also provide an option thru command line to set a higher limit to cover cases where `kMaxDX12BlockLimit` is too small.

Reply:
Should explain in the PR desc why this is desired.

As far as I know this is a pretty rare case. If this causes the API to not be detected then we can tell user to expand the limit.